internal/cluster/clusterdriver/transport_http_test.go

```diff
@@ -40,7 +40,7 @@ func TestRegisterHTTPHandlers_DeleteCluster(t *testing.T) {
 			endpointFunc: func(ctx context.Context, request interface{}) (response interface{}, err error) {
 				return DeleteClusterResponse{Deleted: true}, nil
 			},
-			expectedStatusCode: http.StatusNoContent,
+			expectedStatusCode: http.StatusAccepted,
```

Comment:
**Note**: yeah, this was what I meant with my earlier comment, I would expect the already deleted (as in finished the delete successfully) to be `StatusNoContent` instead, because if there is nothing to do after a finished delete, it should remain the pre-existing value. However if the delete is still in progress and could run into error which would stop the delete process then it should be `StatusAccepted` indeed.

This was what I meant to split the two cases instead of merging them and changing the previous behavior, we need to handle them individually.